distribution/client/src/lib/storage.ts

```diff
@@ -98,6 +103,18 @@ export default class Storage {
         });
 
       }));
+      retArray.push(new Promise((resolve, reject) => {
+
+        fs.unlink(`${pluginPath}/${plugin}.jpi`, (err) => {
+          if (err) {
+            logger.debug(`${pluginPath}/${plugin}.jpi was not found.`);
+          }
+          logger.info(`${pluginPath}/${plugin}.jpi was deleted`);
+          UI.publish(`Deleted ${plugin}.jpi`);
+          resolve(true);
+        });
+
+      }));
```

Comment:
Going to file a followup dedicated PR to promisify it. I didn't want to mix feature addition, and reworking all this (because there are three similar blocks)
https://issues.jenkins-ci.org/browse/JENKINS-54545 to track this.